pymemcache/client/base.py

```diff
@@ -130,6 +135,56 @@ def normalize_server_spec(server):
     return (host, port)
 
 
+def get_positive_integer(param):
+    """
+    Check if the passed param is a positive integer and return an integer
+    from it.
+
+    Zero, negative value, or non numerical values are not allowed.
+    String representation of a number can be passed.
```

Comment:
As we look toward type hints in a future version, I think we want to start being stricter about which types we accept, especially for new code. I therefore think we should only accept integers here (and not strings or floats).

Given that, we can move the positive integer checks into `KeepaliveOpts.__init__` below and remove this helper function.

Reply:
Make sense, however, what do you think about simply moving this function as a method of the `KeepaliveOpts` class rather than moving the check into the constructor?

It will avoid to duplicate the check, it will reduce the technical debt and it will reduce the number of line.

This is a compromising between the previous version and the future version.

Reply:
But it still allows non-`int` values? What is the reason why we need to provide that type conversion for new code?

Reply:
The goal isn't to allow non-int values, the goal is the contrary. Force the type.
We only ensure that we pass the right type of value else it will fail later during the moment when we set the socket option, example:

```
>>> import socket
>>> sock = socket.socket(socket.SOCK_STREAM)
>>> sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_KEEPIDLE, "1")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
```

Ensuring the passed type is right here is cheap. It could be easily removed later when you will enforce strict type.

Else, we could remove my type checking as you suggested and we could rely only on the `TypeError` raised in the example above. It will inform the user that something went wrong. It is more a pythonic way to handle that. I'm not against too.

Thoughts?

Reply:
Yes, I'd rather keep our end of the code minimal and specific for these cases (without the type conversions).

Reply:
Ok no problem, then I'll submit a new patch set soon.

Reply:
Done with fc4df8c